client/extensions/woocommerce/app/promotions/promotion-header.js

```diff
@@ -0,0 +1,83 @@
+/**
+ * External dependencies
+ */
+import React from 'react';
+import PropTypes from 'prop-types';
+import { localize } from 'i18n-calypso';
+import Gridicon from 'gridicons';
+import { isObject } from 'lodash';
+
+/**
+ * Internal dependencies
+ */
+import ActionHeader from 'woocommerce/components/action-header';
+import Button from 'components/button';
+import { getLink } from 'woocommerce/lib/nav-utils';
+
+function renderTrashButton( onTrash, promotion, isBusy, translate ) {
+	return onTrash && (
+		<Button borderless scary onClick={ onTrash }>
+			<Gridicon icon="trash" />
+			<span>{ translate( 'Delete' ) }</span>
+		</Button>
+	);
+}
+
+function renderSaveButton( onSave, promotion, isBusy, translate ) {
+	const saveExists = 'undefined' !== typeof onSave;
```

Comment:
Why not just return immediately if this is false?

Reply:
Good point. Added.